torch/csrc/api/include/torch/nn/options/transformerlayer.h

```diff
@@ -16,7 +16,7 @@ namespace nn {
 /// ```
 struct TORCH_API TransformerEncoderLayerOptions {
 
-  using activation_t = c10::variant<enumtype::kReLU, enumtype::kGELU>;
+  using activation_t = c10::variant<enumtype::kReLU, enumtype::kGELU, std::function<Tensor(const Tensor&)> >;
```

Comment:
Minor - might want to hoist `activation_t` up out of here and `TransformerOptions` and use a single definition.

But on the c10::variant<std::function> thing, are you thinking to try the variant.h fix here?

Reply:
Yep I'll try the variant.h fix here! Thanks for digging into it some more :)

Reply:
@bhosmer Looks like it worked!

Reply:
Ah awesome! thanks for doing it, so satisfying when lego pieces fit together 😁